File: js/serial.js
const fs = require('fs');
const path = require('path');
const shared = require('./shared');

const { bucket, inputFile, output } = shared.getCLIParameters();
const filenames = shared.readFilenames(inputFile);

async function fetchAllSerial(filenames) {
  while (filenames.length > 0) {
    filename = filenames.pop();
    const contents = await shared.fetchFile(bucket, filename);
    await shared.writeFile(path.join(output, filename), contents);
  }
}

if (require.main === module) {
  fetchAllSerial(filenames);
}

File: python/serial.py
from shared import get_params, read_filenames, download_file

params = get_params()
bucket_name = params['bucket_name']
input_file = params['input_file']
output_dir = params['output_dir']

lines = read_filenames(input_file)


def main():
    while(len(lines) > 0):
        filename = lines.pop()
        download_file(bucket_name, filename, output_dir)


if __name__ == '__main__':
    main()
